File: students-backend/src/main/java/com/github/voodu/studentrest/controller/CourseInfoController.java
package com.github.voodu.studentrest.controller;

import com.github.voodu.studentrest.model.CourseInfo;
import com.github.voodu.studentrest.service.CourseInfoService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@CrossOrigin(origins = "*", maxAge = 3600)
@RequestMapping("/public/courses")
public class CourseInfoController {
    @Autowired
    private CourseInfoService courseInfoService;

    @GetMapping //public
    public List<CourseInfo> findAllCourseInfos() {
        return courseInfoService.findAll();
    }

    @GetMapping(value = "/{id}") //public
    public ResponseEntity<CourseInfo> findCourseInfo(@PathVariable long id) {
        CourseInfo found =  courseInfoService.getOne(id);
        if (found != null)
        {
            return new ResponseEntity<>(found, HttpStatus.OK);
        }

        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
    }

    @PostMapping //only admin
    public CourseInfo addCourseInfo(@RequestBody CourseInfo courseInfo) {
        courseInfo.id = null;
        return courseInfoService.save(courseInfo);
    }

    @PutMapping //only admin
    public ResponseEntity<CourseInfo> putCourseInfo(@RequestBody CourseInfo courseInfo) {
        CourseInfo saved = courseInfoService.update(courseInfo);
        if (saved != null)
        {
            return new ResponseEntity<>(saved, HttpStatus.OK);
        }

        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
    }

    @DeleteMapping(value = "/{id}") //only admin
    public ResponseEntity<CourseInfo> deleteCourseInfo(@PathVariable("id") long id) {
        if (courseInfoService.deleteById(id)) {
            return new ResponseEntity<>(HttpStatus.OK);
        }

        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
    }
}

File: students-backend/src/main/java/com/github/voodu/studentrest/controller/CourseInfoController.kt
package com.github.voodu.studentrest.controller

import com.github.voodu.studentrest.model.CourseInfo
import com.github.voodu.studentrest.service.AppUserService
import com.github.voodu.studentrest.service.CourseInfoService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.RequestEntity
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import javax.servlet.http.HttpServletRequest

@RestController
@CrossOrigin(origins = arrayOf("*"), maxAge = 3600)
@RequestMapping("/public/courses")

class CourseInfoController @Autowired constructor(appUserService: AppUserService, private val courseInfoService: CourseInfoService): AuthorizedController(appUserService) {

    @GetMapping //public
    fun findAllCoursesInfo(request: HttpServletRequest) = whenAuthorized(3, request) {
        ResponseEntity.ok(courseInfoService.findAll())
    }

    @GetMapping(value = ["/{id}"]) //public
    fun findCourseInfo(@PathVariable id: Long, request: HttpServletRequest) = whenAuthorized(3, request){
        val found = courseInfoService.getOne(id)
        if (found != null) {
            ResponseEntity.ok(found)
        } else {
            ResponseEntity(HttpStatus.NOT_FOUND)
        }
    }

    @PostMapping //only admin
    fun addCourseInfo(@RequestBody courseInfo: CourseInfo, request: HttpServletRequest) = whenAuthorized(0, request) {
        ResponseEntity.ok(courseInfoService.save(courseInfo))
    }

    @PutMapping //only admin
    fun putCourseInfo(@RequestBody courseInfo: CourseInfo, request: HttpServletRequest) = whenAuthorized(0, request) {
        val saved = courseInfoService.update(courseInfo)
        if (saved != null) {
            ResponseEntity.ok(saved)
        } else {
            ResponseEntity(HttpStatus.NOT_FOUND)
        }

    }

    @DeleteMapping(value = ["/{id}"]) //only admin
    fun deleteCourseInfo(@PathVariable("id") id: Long, request: HttpServletRequest) = whenAuthorized(0, request) {
        val deleted = courseInfoService.deleteById(id)
        if (deleted != null) {
            ResponseEntity.ok(deleted)
        } else {
            ResponseEntity(HttpStatus.NOT_FOUND)
        }
    }


}
